protocols/mdns/src/behaviour.rs

```diff
@@ -359,13 +368,19 @@ where
             } else {
                 tracing::info!(%peer, address=%addr, "discovered peer on address");
                 self.discovered_nodes.push((peer, addr.clone(), expiration));
-                discovered.push((peer, addr));
+                discovered.push((peer, addr.clone()));
+
+                self.pending_events
+                    .push_back(ToSwarm::NewExternalAddrOfPeer {
+                        peer_id: peer,
+                        address: addr,
+                    });
             }
         }
 
         if !discovered.is_empty() {
             let event = Event::Discovered(discovered);
-            return Poll::Ready(ToSwarm::GenerateEvent(event));
+            self.pending_events.push_back(ToSwarm::GenerateEvent(event));
```

Comment:
```suggestion
            self.pending_events.push_back(ToSwarm::GenerateEvent(event));
            continue;
```

Otherwise the `Event:Expired` below will be returned before the items in `pending_events`.

Reply:
```suggestion
            self.pending_events.push_front(ToSwarm::GenerateEvent(event));
            continue;
```

Actually, IMO it's more intuitive to first report the mdns event, and then the new external addresses. 

We do the same pattern in identify: first report the identify `Received` event for the user, then the extracted `NewExternalAddrOfPeer` info for other behaviors.

Reply:
In that case the expired events would be delayed because of the events in the pending_events right? again I may be wrong here

Reply:
You mean if we call `continue` here?

It's the same logic as the current master, where the `Discovered` event is returned before handling any expired events.